src/routes/landing/components/FilterableCitySelector.js

```diff
@@ -17,34 +19,70 @@ type PropsType = {
 }
 
 type StateType = {
-  filterText: string
+  filterText: string,
+  currentLongitude: number | null,
+  currentLatitude: number | null,
+  renderLocationList: boolean
 }
 
 class FilterableCitySelector extends React.Component<PropsType, StateType> {
   constructor (props: PropsType) {
     super(props)
-    this.state = { filterText: '' }
+    this.state = {
+      filterText: '',
+      currentLongitude: null,
+      currentLatitude: null,
+      renderLocationList: false
+    }
   }
 
   onFilterTextChange = (filterText: string) => this.setState({ filterText })
 
+  _onPressLocationButton = () => {
+    if (Platform.OS === 'ios') {
+      Geolocation.requestAuthorization()
```

Comment:
Why is this only needed on ios?

Reply:
On Android the request happens automatically, when you use Geolocation.getCurrentPosition()